source/code/providers/support/mycnf-scanner.h

```diff
@@ -0,0 +1,155 @@
+#include <set>
+#include <map>
+
+#include <scxcorelib/scxdirectoryinfo.h>
+#include <scxcorelib/scxexception.h>
+#include <scxcorelib/scxfile.h>
+#include <scxcorelib/scxfilepath.h>
+#include <scxcorelib/scxfilesystem.h>
+#include <scxcorelib/scxhandle.h>
+#include <scxcorelib/scxlog.h>
+#include <scxcorelib/stringaid.h>
+
+#include "sqlauth.h"
+
+namespace MySQL  {
+
+   /**
+      MySQL Configuration Scanner dependencies class
+
+      Allows overriding for unit test purposes
+   */
+    class MySQL_CnfScannerDependency
+    {
+    public:
+        //! Ctor
+        MySQL_CnfScannerDependency() { }
+        //! Dtor
+        virtual ~MySQL_CnfScannerDependency() {}
+        
+        /**
+       Finds the paths of file system objects in a specified directory.
+       
+       \param[in] path	   Path to directory.
+       
+       \returns vector containing the files in this directory
+       
+       \throws SCXFilePathNotFoundException if path is not found.
+        */
+        virtual std::vector< SCXCoreLib::SCXFilePath >  GetFiles(const SCXCoreLib::SCXFilePath& path)
+        {
+            return SCXCoreLib::SCXDirectory::GetFileSystemEntries(path, SCXCoreLib::eDirSearchOptionFile) ;
+        }
+
+        /**
+           Reads as many lines of the UT8 encoded file at the specified position
+           in the file system as possible.
+           
+           Handles newline symbols in a platform independent way, that is, may
+           be used to read a stream originating on one platform on another platform.
+           If the lines is to be written back to the originating system
+           the same nlf should in general be used, if we do not have other information.
+           
+           \param[in]  source  Position of the file in the filesystem
+           \param[in]  lines   Lines that was read
+           \param[out] nlfs	  Newline symbols found used in the file
+           \throws SCXUnauthorizedFileSystemAccessException
+           \throws SCXLineStreamPartialReadException	 Line to long to be stored in a wstring
+           \throws SCXLineStreamContentException		 Invalid byte sequence according to UTF8
+           
+           \note Will return no lines if file not found.
+        */
+        virtual void ReadAllLinesAsUTF8(
+            const SCXCoreLib::SCXFilePath & source,
+            std::vector< std::wstring >   & lines,
+            SCXCoreLib::SCXStream::NLFs   & nlfs)
+        {
+            SCXCoreLib::SCXFile::ReadAllLinesAsUTF8(source, lines, nlfs);
+        }
+
+    private:
+    };
+
+   /**
+   MySQL Configuration Scanner Class
+
+   flattens the given config file into a vector of wstrings by acting on '!include' and '!includedir' directives
+   */
+   class MySQL_ConfigurationScanner
+   {
+   public:
+
+       MySQL_ConfigurationScanner()
+           : m_extDepends(new MySQL_CnfScannerDependency()),
+             m_log(SCXCoreLib::SCXLogHandleFactory::GetLogHandle(L"mysql.cwconfigurationscanner"))
```

Comment:
My bad.  I will fix it. (The one time I use  vi, world will get to know of it :cry):